Assets/Scripts/Strategy/Transitions/BattleResolver.cs

```diff
@@ -16,6 +17,18 @@ private void Awake()
             ISquad squad = new Squad(SceneSharing.squadID);
             if (GameScenes.battleWin)
             {
+
+                foreach(Town town in Town.GetAllTowns())
+                {
+                    foreach(EnemySquad enemy in EnemySquad.GetAllEnemySquads())
+                    {
+                        if (squad.X == town.X && squad.Y == town.Y && squad.X == enemy.X && squad.Y == enemy.Y)
+                        {
+                            town.PlayerOwned = true;
+                        }
+                    }
```

Comment:
You would only need to check if a squad is in a town correct? And it should be set to whether it was a win by the player or not